### PR TITLE
Provisioner state address

### PR DESCRIPTION
**Pull request by wallyworld**
The provisioner worker was calling a StateAddresses() API but it doesn't need that info. Remove the API and fix the associated fallout.

The second commit is a drive by to move some trivial container helper methods from state to core. This removes one of the cases where the worker package imports state. Sadly it's one of many, so there's more to do.

## QA steps

bootstrap k8s and deploy a charm
juju show-controller and check the api address is correct

bootstrap lxd and deploy a charm
juju show-controller and check the api address is correct
juju enable-ha
juju show-controller and check the api addresses are correct
remove a controller machine
enable-ha again
check that a replacement controller comes up and show-controller is happy

## Bug reference

https://bugs.launchpad.net/juju/+bug/1866643

### CODE REVIEW

**Review by ycliuhw — APPROVED** [review]
lgtm thanks

**Comment by wallyworld** [comment]
[Provisioner shouldn't need to know StateAddresses](https://trello.com/c/HCZ0OmX8/1233-provisioner-shouldnt-need-to-know-stateaddresses)

**Comment by wallyworld** [comment]
$$merge$$

File: apiserver/facades/agent/provisioner/provisioner_test.go [hunk 1]
```diff
@@ -1860,17 +1860,6 @@ func (s *withControllerSuite) TestAPIAddresses(c *gc.C) {
 	})
 }
 
-func (s *withControllerSuite) TestStateAddresses(c *gc.C) {
```
Comment (manadart) [inline apiserver/facades/agent/provisioner/provisioner_test.go:1863]: This breaks backward compatibility, even if unused.
I think the version needs to be bumped and the method stubbed for old versions as was done in:
https://github.com/juju/juju/pull/11706

**Pull request merged**